src/irq.rs

```diff
@@ -143,7 +145,7 @@ where
                 self.send_packet();
             }
             State::RampUpTx => {
-                debug_assert!(ready_event);
+                debug_assert!(ready, "RampUpTx de: {}, re: {}, te: {}", disabled, ready, timer);
```

Comment:
We could also impl debug on the `Events` struct, and reduce some of this repeated code.

Then you'd probably want `evts.ready`, instead of the destructuring I do above.